### PR TITLE
Measuring the proper sizes of images in chat

### LINKED ISSUE #15654: Vertical gifs are cropped on mobile devices
# Bug Report

## Problem 
Status Desktop 2023-04-14 12-35-31

#### Expected behavior
can see full-size gif

### Reproduction

1) Open Status
2) Send vertical gif, i.e. [ttps://media.tenor.com/YzwPJ_PVJOsAAAAC/cat-poor.gif](https://media.tenor.com/YzwPJ_PVJOsAAAAC/cat-poor.gif)

### Additional Information

- Status version: nightly 14/04/2023
- Operating System: Android, iOS

[comment]: # (Please, add logs/notes if necessary)

### PR DESCRIPTION
**Pull request by alwx**
Fixes https://github.com/status-im/status-mobile/issues/15654

#### Platforms

- Android
- iOS

##### Functional

- 1-1 chats
- public chats
- group chats

status: ready

### CODE REVIEW

**Review by ibrkhalil — APPROVED** [review]

File: src/status_im2/contexts/chat/messages/link_preview/view.cljs [hunk 1]
```diff
@@ -89,45 +89,54 @@
...
+                   [rn/view (style/separator)]
+                   [fast-image/fast-image
+                    {:source              {:uri thumbnail-url}
+                     :on-load             (fn [e]
+                                            (let [{:keys [width height]} (js->clj (.-nativeEvent e)
+                                                                                  :keywordize-keys
+                                                                                  true)]
+                                              (reset! measured-width width)
+                                              (reset! measured-height height)))
+                     :style               (style/image {:width  @measured-width
```
Comment (siddarthkay) [inline src/status_im2/contexts/chat/messages/link_preview/view.cljs:137]: Are there any safeguards in place if actual width and actual height are greater than our viewport?

Comment (ibrkhalil) [reply src/status_im2/contexts/chat/messages/link_preview/view.cljs:137]: Good point, Should we default to a factor of the screen dimensions ?
Like for example the max or the undefined value to be %25 width and height of the screen

Comment (siddarthkay) [reply src/status_im2/contexts/chat/messages/link_preview/view.cljs:137]: for small images and images with weird aspect ratios we will face scaling issues. I think better is to use object-fit css, something along the lines of this : https://stackoverflow.com/a/34713809 but then again I am not 100% sure of this solution either. 

Will have to spend some time on this to get an optimal state.. @ibrkhalil

Comment (alwx) [reply src/status_im2/contexts/chat/messages/link_preview/view.cljs:137]: There is already a safeguard, this is how the style for `link-preview-loader` is defined:
```
(defn image
  [{:keys [height width] :as dimensions}]
  (merge (if (and (pos? height)
                  (pos? width))
           (scale-dimensions dimensions)
           {:height 170})
         {:overflow      :hidden
          :border-radius 12}))
```
It uses the function `scale-dimension` which scales the pic and takes the screen width into the consideration (see `status-im2.contexts.chat.messages.link-preview.style/scale-dimensions`)

**Review by siddarthkay — APPROVED** [review]
Looks good to me.

**Comment by qoqobolo** [comment]
Thanks for your work @alwx!
PR is tested and can be merged.

**Pull request merged**